src/source/Ice/ConnectionListener.c

```diff
@@ -276,24 +287,16 @@ PVOID connectionListenerReceiveDataRoutine(PVOID arg)
             }
         }
 
-        // Should be one more than the sockets count per API documentation
-        nfds++;
-
         // Need to unlock the mutex to ensure other racing threads unblock
         MUTEX_UNLOCK(pConnectionListener->lock);
 
-        // timeout select every SOCKET_WAIT_FOR_DATA_TIMEOUT_SECONDS seconds and check if terminate
-        // on linux tv need to be reinitialized after select is done.
-        tv.tv_sec = 0;
-        tv.tv_usec = CONNECTION_LISTENER_SOCKET_WAIT_FOR_DATA_TIMEOUT / HUNDREDS_OF_NANOS_IN_A_MICROSECOND;
-
         // blocking call until resolves as a timeout, an error, a signal or data received
-        retval = select(nfds, &rfds, NULL, NULL, &tv);
+        retval = POLL(rfds, nfds, CONNECTION_LISTENER_SOCKET_WAIT_FOR_DATA_TIMEOUT / HUNDREDS_OF_NANOS_IN_A_MILLISECOND);
```

Comment:
Might want to play with the value of CONNECTION_LISTENER_SOCKET_WAIT_FOR_DATA_TIMEOUT as it will impact the latency. Unfortunately, selecting very fast polling will impact the CPU/battery life while using slow polling will impact the latency. If it's actually decided to go with polling then I would imagine this number to be around 40ms

Reply:
`poll` is a misnomer i think
internally it functions in the very same way as `select` except it has better designed input parameters
From linux man pages:
```
DESCRIPTION
       poll()  performs a similar task to select(2): it waits for one of a set of file descriptors to become ready to per‐
       form I/O.
```

Reply:
@MushMal 
> We tried to avoid polling due to inherent impact on latency and battery life.

See above
`poll` doesn't really poll :) it `select`s but with a nicer arguments :)

> I am curious as to how prevalent this case is?

well it is considered a bug to set FD_SET if localSocket int value is over 1024
so if it so happens that there are a lot of open file descriptors, the library wont function at all

Reply:
Familiarized myself with the API. I don't have any further arguments - poll it is!